src/utils/walletMigrationArchanova.js

```diff
@@ -0,0 +1,299 @@
+  if (!fee) {
+    reportErrorLog('submitMigrationTransactions: estimateMigrationTransactions returned falsy fee');
+    throw new Error(t('error.failedToSubmitTransaction'));
+  }
+
+  const tokensToMigrateAfterFee = getTokensToMigrateAfterFee(tokensToMigrate, walletBalances, fee);
+
+  logBreadcrumb('walletMigrationArchanova', 'build transaction to submit', {
+    tokensToMigrateAfterFee,
+    collectiblesToMigrate,
+  });
+  const transactionsToSubmit = await buildAssetMigrationRawTransactions(
+    wallet,
+    accounts,
+    tokensToMigrateAfterFee,
+    collectiblesToMigrate,
+  );
+  if (!transactionsToSubmit?.length) {
+    reportErrorLog('submitMigrationTransactions: transactionsToSumbmit is empty');
+    throw new Error(t('error.failedToSubmitTransaction'));
+  }
+
+  let hash;
+  try {
+    logBreadcrumb('walletMigrationArchanova', 'send raw transaction');
+    hash = await archanovaService.sendRawTransactions(transactionsToSubmit);
+  } catch (error) {
+    reportErrorLog('submitMigrationTransactions: sendRawTransactions error', { error });
+    throw new Error(t('error.failedToSubmitTransaction'));
+  }
+
+  if (!hash) {
+    reportErrorLog('submitMigrationTransactions: sendRawTransactions returned falsy hash');
+    throw new Error(t('error.failedToSubmitTransaction'));
+  }
+
+  logBreadcrumb('walletMigrationArchanova', 'transaction submitted', { hash });
+  return hash;
+}
+
+/**
+ * High-level function responsible for estimating gas fee for migration transactions.
+ */
+export async function estimateMigrationTransactions(
+  wallet: Wallet,
+  accounts: Account[],
+  tokensToMigrate: TokensToMigrateByAddress,
+  collectiblesToMigrate: CollectibleId[],
+): Promise<BigNumber> {
+  logBreadcrumb('walletMigrationArchanova', 'building asset migration raw transactions', {
+    accounts,
+    tokensToMigrate,
+    collectiblesToMigrate,
+  });
+  const transactionsToEstimate = await buildAssetMigrationRawTransactions(
+    wallet,
+    accounts,
+    tokensToMigrate,
+    collectiblesToMigrate,
+  );
+  if (!transactionsToEstimate?.length) {
+    reportErrorLog('submitMigrationTransactions: transactionsToEstimate is empty');
+    throw new Error(t('error.failedToEstimateTransaction'));
+  }
+
+  logBreadcrumb('walletMigrationArchanova', 'estimate archanova raw transaction');
+  const fee = await estimateArchanovaRawTransactions(transactionsToEstimate);
+
+  logBreadcrumb('walletMigrationArchanova', 'estimated archanova raw transaction', { fee: fee.toFixed() });
+  return fee;
+}
+
+export function hasNonNegligileWalletBalances(balances: ?WalletAssetsBalances, rates: RatesByAssetAddress) {
+  if (!balances) return false;
+
+  return Object.keys(balances).some((symbol) => isNonNegligibleBalance(balances[symbol], rates));
+}
+
+const NON_NEGLIGIBLE_BALANCE_THRESHOLD_IN_TOKEN = '0.25';
+const NON_NEGLIGIBLE_BALANCE_THRESHOLD_IN_USD = 1;
+
+export function isNonNegligibleBalance(balance: WalletAssetBalance, rates: RatesByAssetAddress) {
```

Comment:
Maybe `isEnoughBalanceToMigrate`?

Reply:
Please check [this comment](https://github.com/pillarwallet/pillarwallet/pull/2784#discussion_r679360435).